atc/api/accessor/role_action_map.go

```diff
@@ -0,0 +1,129 @@
+	atc.CheckResource:                 "pipeline-operator",
+	atc.CheckResourceWebHook:          "pipeline-operator",
+	atc.CheckResourceType:             "pipeline-operator",
+	atc.ListResourceVersions:          "viewer",
+	atc.GetResourceVersion:            "viewer",
+	atc.EnableResourceVersion:         "pipeline-operator",
+	atc.DisableResourceVersion:        "pipeline-operator",
+	atc.PinResourceVersion:            "pipeline-operator",
+	atc.ListBuildsWithVersionAsInput:  "viewer",
+	atc.ListBuildsWithVersionAsOutput: "viewer",
+	atc.GetResourceCausality:          "viewer",
+	atc.ListAllPipelines:              "viewer",
+	atc.ListPipelines:                 "viewer",
+	atc.GetPipeline:                   "viewer",
+	atc.DeletePipeline:                "member",
+	atc.OrderPipelines:                "member",
+	atc.PausePipeline:                 "pipeline-operator",
+	atc.UnpausePipeline:               "pipeline-operator",
+	atc.ExposePipeline:                "member",
+	atc.HidePipeline:                  "member",
+	atc.RenamePipeline:                "member",
+	atc.ListPipelineBuilds:            "viewer",
+	atc.CreatePipelineBuild:           "member",
+	atc.PipelineBadge:                 "viewer",
+	atc.RegisterWorker:                "member",
+	atc.LandWorker:                    "member",
+	atc.RetireWorker:                  "member",
+	atc.PruneWorker:                   "member",
+	atc.HeartbeatWorker:               "member",
+	atc.ListWorkers:                   "viewer",
+	atc.DeleteWorker:                  "member",
+	atc.SetLogLevel:                   "member",
+	atc.GetLogLevel:                   "viewer",
+	atc.DownloadCLI:                   "viewer",
+	atc.GetInfo:                       "viewer",
+	atc.GetInfoCreds:                  "viewer",
+	atc.ListContainers:                "viewer",
+	atc.GetContainer:                  "viewer",
+	atc.HijackContainer:               "member",
+	atc.ListDestroyingContainers:      "viewer",
+	atc.ReportWorkerContainers:        "member",
+	atc.ListVolumes:                   "viewer",
+	atc.ListDestroyingVolumes:         "viewer",
+	atc.ReportWorkerVolumes:           "member",
+	atc.ListTeams:                     "viewer",
+	atc.GetTeam:                       "viewer",
+	atc.SetTeam:                       "owner",
+	atc.RenameTeam:                    "owner",
+	atc.DestroyTeam:                   "owner",
+	atc.ListTeamBuilds:                "viewer",
+	atc.CreateArtifact:                "member",
+	atc.GetArtifact:                   "member",
+	atc.ListBuildArtifacts:            "viewer",
+}
+
+type CustomActionRoleMap map[string][]string
+
+//go:generate counterfeiter . ActionRoleMap
+
+type ActionRoleMap interface {
+	RoleOfAction(string) string
+}
+
+//go:generate counterfeiter . ActionRoleMapModifier
+
+type ActionRoleMapModifier interface {
+	CustomizeActionRoleMap(lager.Logger, CustomActionRoleMap) error
+}
+
+func ParseCustomActionRoleMap(filename string, mapping *CustomActionRoleMap) error {
```

Comment:
hi @evanchaoli, the file parsing logic here is similar to how we validate and read from files in [this flags library](https://github.com/concourse/flag/blob/master/file.go). To use it, we input config values as `flag.File` rather than `string`, like this:

https://github.com/concourse/concourse/blob/84acaa94d6d7bf9bfd895770ede79ceb1c8dd6d4/atc/atccmd/command.go#L96

Another advantage of parsing inputs this way is that the existence of the file is validated earlier. Would you be up for refactoring this later on to leverage the existing library method? Let us know if you need support.